compiler-rt/test/asan/TestCases/Windows/allocator_may_return_null_set_via_user_function.cpp

```diff
@@ -0,0 +1,27 @@
+// RUN: %clangxx_asan -O0 %s -o %t
+// RUN: %run %t 2>&1
+// CHECK: Success
+
+#include <cstdint>
+#include <cstdio>
+#include <cstdlib>
+#include <limits>
+
+// On Windows, flags configured through the user-defined function `__asan_default_options`
```

Comment:
This and allocator_may_return_null_limits.cpp are really variations of the same test. I wonder if they could be folded together?

For example, the` __asan_default_options()` definition could be guarded by an `#ifdef` and then you could run multiple invocations on the same file. Something like:

```
// RUN: %clangxx_asan -DASAN_DEFAULT_OPTIONS %s -o %t
// RUN: %run %t 2>&1 | FileCheck --check-prefix=RETURN_NULL
// RUN: %clangxx_asan %s -o %t
// RUN: %env_asan_opts=allocator_may_return_null=1 %run %t 2>&1 | FileCheck --check-prefix=RETURN_NULL
// RUN: %env_asan_opts=allocator_may_return_null=0 not %run %t 2>&1 | FileCheck --check-prefix=ABORT
```

Reply:
Yup, good idea! I actually had a local branch where I'm tackling #117925 more generally, and I had already merged the tests there. So I'm porting that over, here: https://github.com/llvm/llvm-project/pull/117929/commits/ebc3d0df3d55f1e03add8007970042b566954d32